File: frontend/types/roadmap.ts
export interface RoadmapResource {
    name: string;
    url: string;
    type: string;
    cost: "Free" | "Paid";
    description?: string;
    is_local: boolean;
    thumbnail_url?: string;
}

export interface RoadmapMonth {
    month: number;
    title: string;
    skills: string[];
    tasks: string[];
    projects: string[];
    detailed_guide?: string;
    resources: RoadmapResource[];
}

export interface DailyTask {
    day: string;
    topic: string;
    duration_min: number;
    reminder_text: string;
}

export interface ExecutionSprint {
    week_number: number;
    milestone_title: string;
    days: DailyTask[];
    focus_area: string;
}

export interface ExecutionSchedule {
    sprints: ExecutionSprint[];
    active_sprint_index: number;
}

export interface RoadmapOutput {
    summary: string;
    months: RoadmapMonth[];
    additional_info?: string;
    execution_schedule?: ExecutionSchedule;
    target_role: string;
    hours_per_week: number;
    timeframe_months?: number;
    location?: string;
    logs?: {
        timestamp: string;
        source: string;
        message: string;
        metadata?: any;
    }[];
}

export interface RoadmapInput {
    name?: string;
    location: string;
    current_status: "student" | "graduate" | "junior dev";
    degree?: string;
    skills: string[];
    skill_level: "beginner" | "intermediate" | "advanced";
    target_role: string;
    hours_per_week: number;
    timeframe_months: number;
    constraints: string[];
}

File: backend/app/schemas/roadmap.py
from pydantic import BaseModel, Field
from typing import List, Optional

class RoadmapResource(BaseModel):
    name: str
    url: str
    type: str # e.g., "course", "article", "community", "video"
    cost: str # "Free" or "Paid"
    description: Optional[str] = None
    is_local: bool = False
    thumbnail_url: Optional[str] = None

class RoadmapMonth(BaseModel):
    month: int
    title: str
    skills: List[str]
    tasks: List[str]
    projects: List[str]
    detailed_guide: Optional[str] = None # Markdown content for deep dive
    resources: List[RoadmapResource]

class RoadmapSection(BaseModel):
    title: str
    content: str

class DailyTask(BaseModel):
    day: str
    topic: str
    duration_min: int
    reminder_text: str

class ExecutionSprint(BaseModel):
    week_number: int
    milestone_title: str
    days: List[DailyTask]
    focus_area: str

class ExecutionSchedule(BaseModel):
    sprints: List[ExecutionSprint]
    active_sprint_index: int = 0

class RoadmapOutput(BaseModel):
    summary: str
    months: List[RoadmapMonth]
    additional_info: Optional[str] = None
    execution_schedule: Optional[ExecutionSchedule] = None

class RoadmapInput(BaseModel):
    name: Optional[str] = None
    location: str = "Kenya"
    current_status: str # student, graduate, junior dev
    degree: Optional[str] = None
    skills: List[str]
    skill_level: str # beginner, intermediate, advanced
    target_role: str
    hours_per_week: int
    timeframe_months: int # 3 or 6
    constraints: List[str] # e.g., "laptop", "4G bundles"
